compiler/noirc_frontend/src/elaborator/scope.rs

```diff
@@ -42,15 +43,36 @@ impl<'context> Elaborator<'context> {
         ModuleId { krate: self.crate_id, local_id: self.local_module }
     }
 
-    pub(super) fn resolve_path(&mut self, path: Path) -> Result<ModuleDefId, ResolverError> {
+    pub(super) fn resolve_path_or_error(
+        &mut self,
+        path: Path,
+    ) -> Result<ModuleDefId, ResolverError> {
+        let path_resolution = self.resolve_path(path)?;
+
+        if let Some(error) = path_resolution.error {
+            self.push_err(error);
+        }
+
+        Ok(path_resolution.module_def_id)
+    }
+
+    pub(super) fn resolve_path(&mut self, path: Path) -> PathResolutionResult {
         let mut module_id = self.module_id();
         let mut path = path;
 
+        if path.kind == PathKind::Plain {
+            self.def_maps.get_mut(&self.crate_id).unwrap().modules[module_id.local_id.0]
+                .use_import(&path.segments[0].ident);
```

Comment:
```suggestion
            let module_data =
                &mut self.def_maps.get_mut(&self.crate_id).unwrap().modules[module_id.local_id.0];
            module_data.use_import(&path.segments[0].ident);
```

Small nit but I'd split this line up